From d8e2f1cc6c94b4b7c5c80c7f46b709dddb1e5369 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 25 Aug 2015 16:29:32 -0700
Subject: [PATCH] Fix compile errors on mingw

---
 Makefile.libretro |  2 +-
 src/util/common.h | 10 ++++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index ffb9e80..0ae93a2 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -254,7 +254,7 @@ else
 	CC = gcc
 	CXX = g++
 	SHARED := -shared -static-libgcc -static-libstdc++ -Wl,-no-undefined -Wl,-version-script=link.T
-	DEFINES += -std=c99 -DDISABLE_THREADING -DGIT_VERSION=\"$(GIT_VERSION)\"
+	DEFINES += -std=c99 -DDISABLE_THREADING -DGIT_VERSION=\"$(GIT_VERSION)\" -Doff_t=_off_t -Doff64_t=_off64_t
 
 endif
 
diff --git a/src/util/common.h b/src/util/common.h
index 13e8003..fea28cd 100644
--- a/src/util/common.h
+++ b/src/util/common.h
@@ -32,6 +32,16 @@ typedef intptr_t ssize_t;
 #include <unistd.h>
 #endif
 
+#ifdef __MINGW32__
+#define strcasecmp _stricmp
+#define strncasecmp _strnicmp
+#if defined(__STRICT_ANSI__)
+#undef __STRICT_ANSI__
+#include <sys/stat.h>
+#define __STRICT_ANSI__
+#endif
+#endif
+
 #ifndef SSIZE_MAX
 #define SSIZE_MAX ((ssize_t) (SIZE_MAX >> 1))
 #endif
-- 
1.9.5.msysgit.0

